SwiftCompilerSources/Sources/Optimizer/FunctionPasses/ClosureSpecialization.swift

```diff
@@ -762,6 +770,34 @@ private extension PartialApplyInst {
   }
 }
 
+extension EnumInst {
+  public var caseName: StringRef {
+    return self.type.getEnumCases(in: self.parentFunction)![self.caseIndex]!.name
+  }
+
+  public var isOptionalSome: Bool {
+    assert(self.type.isOptional)
+    assert(caseName == "none" || caseName == "some")
+    return caseName == "some"
```

Comment:
It's more efficient to compare the case index against `Builder.optionalNoneCaseIndex`/`Builder.optionalSomeCaseIndex`

Reply:
Switched from comparing case names to comparing case indexes, thanks - see c2a14cbf31f9bfe2888ae0cfbbaf9a97fcb0d081